src/config/enhance-octokit.ts

```diff
@@ -1,26 +1,31 @@
-import OctokitError from "../models/octokit-error";
 import statsd from "./statsd";
 import { extractPath } from "../jira/client/axios";
 import { GitHubAPI } from "probot";
 import { metricHttpRequest } from "./metric-names";
+import { getLogger } from "./logger";
+import { Octokit } from "@octokit/rest";
+
+const logger = getLogger("octokit");
 
 const instrumentRequests = (octokit: GitHubAPI) => {
 	octokit.hook.wrap("request", async (request, options) => {
 		const requestStart = Date.now();
 		let responseStatus = null;
 
+		let response:Octokit.Response<any>;
+		let error:any;
 		try {
-			const response = await request(options);
+			response = await request(options);
 			responseStatus = response.status;
-
 			return response;
-		} catch (error) {
-			if (error.responseCode) {
-				responseStatus = error.responseCode;
-			}
-
+		} catch (err) {
+			error = err;
+			responseStatus = error?.responseCode;
 			throw error;
 		} finally {
+			if(error || responseStatus < 200 || responseStatus >= 400) {
+				logger.warn({request, response, error}, `Octokit error: '${responseStatus}' for request '${options.method} ${options.url}'`);
```

Comment:
I wonder if we have a logger in request.context at this point? So we will have requestId (or job id) for tracing?

Reply:
this is for the github api, not express.  There is no `req.context` for it.

Reply:
I mean, it would be great if we could pass it to the request where we make a call. And here use (request.log || logger).warn

Reply:
However if now we need a quick win to see errors then we can take care of it later.